common/oauth.go

```diff
@@ -11,3 +11,20 @@ type OauthContext struct {
 	RefreshToken string
 	Refresher    TokenRefresher
 }
+
+// OAuthExchange is the API message for exchange OAuth context.
+type OAuthExchange struct {
+	ClientID     string
+	ClientSecret string
+}
+
+// OAuthToken is the API message for OAuthToken.
+type OAuthToken struct {
+	AccessToken  string `json:"access_token" jsonapi:"attr,accessToken"`
+	RefreshToken string `json:"refresh_token" jsonapi:"attr,refreshToken"`
+	ExpiresIn    int64  `json:"expires_in"`
+	CreatedAt    int64  `json:"created_at"`
+	// ExpiresTs is a derivative from ExpresIn and CreatedAt.
+	// ExpiresTs = ExpiresIn == 0 ? 0 : CreatedAt + ExpiresIn
+	ExpiresTs int64 `jsonapi:"attr,expiresTs"`
```

Comment:
1st, in the common module, jsonapi should never appear.

2nd, can we double check if we do need to return expiresTs to the frontend? Looks like it's only used by the retry logic in the backend https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/plugin/vcs/gitlab/gitlab.go?L765

Reply:
we may still need this field when creating a repository

Reply:
https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/frontend/src/store/modules/repository.ts?L91

Reply:
I see, we can refactor it later.

but we should use json tag instead of jsonapi